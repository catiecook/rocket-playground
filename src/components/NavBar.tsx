import React from 'react';
import Nav from 'react-bootstrap/Nav';

function NavBar() {
  return (
    <Nav className="nav">
      <Nav.Item>
        <Nav.Link href="#/">Home</Nav.Link>
      </Nav.Item>
      <Nav.Item>
        <Nav.Link href="#/graphs">Graphs</Nav.Link>
      </Nav.Item>
    </Nav>
  );
}

export default NavBar;
