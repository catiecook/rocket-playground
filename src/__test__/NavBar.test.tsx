import React from "react";
import ReactDOM from 'react-dom';
import { fireEvent, waitForElement } from "@testing-library/react";
import Enzyme, { shallow, render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import NavBar from "../components/NavBar";

Enzyme.configure({ adapter: new Adapter() })

it('NavBar rendrs as expected', () => {
  const wrapper = shallow(<NavBar />)
  expect(toJson(wrapper)).toMatchSnapshot();
});

describe("<NavBar />", () => {
  test("renders NavBar view", async () => {
    const wrapper = shallow(<NavBar />)
    expect(wrapper.find("Nav"));
  });
});
