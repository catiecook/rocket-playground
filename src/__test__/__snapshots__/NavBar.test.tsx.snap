// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar rendrs as expected 1`] = `
<Nav
  className="nav"
  fill={false}
  justify={false}
>
  <NavItem>
    <NavLink
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        }
      }
      disabled={false}
      href="#/"
    >
      Home
    </NavLink>
  </NavItem>
  <NavItem>
    <NavLink
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        }
      }
      disabled={false}
      href="#/graphs"
    >
      Graphs
    </NavLink>
  </NavItem>
  <NavItem>
    <NavLink
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        }
      }
      disabled={false}
      href="#/rocket-image"
    >
      Rocket Image
    </NavLink>
  </NavItem>
</Nav>
`;
